content/projects/gsoc/2021/project-ideas/tekton-client-plugin.adoc

```diff
@@ -1,47 +1,48 @@
 ---
 layout: gsocprojectidea
-title: "Tekton client plugin for Jenkins"
-goal: "Trigger Tekton Pipelines from Jenkins Jobs."
+title: "Tekton Pipelines DSL for Jenkins with the Tekton Client Plugin"
+goal: "Manage Tekton resources via Pipeline Steps / Jenkinsfile"
 category: Plugins
 year: 2021
 status: draft
 sig: docs
 skills:
 - Java
 - Kubernetes
-- Tekton
+- Tekton Pipelines
 mentors:
 - "waveywaves"
 - "garethjevans"
 links:
   draft: https://docs.google.com/document/d/1_mS0gLgM7oz7PanE0X_SkJ8iUo-BPVCixXMSDO7-VMk/edit#
+  design doc: https://docs.google.com/document/d/10n3vgYrjS_M1571OUOYLw4tK2QmxY7kp_wyUT3ke4gY/edit#
 ---
 
 === Background
 With the movement of traditional software architecture to microservices, there has been a growth in usage of container orchestration infrastructure (Kubernetes) and tooling around it. To adapt to this there have been plugins developed for Jenkins like the kubernetes-plugin which allow this transition to be smooth for the users to keep the same CI/CD platform and migrate their workloads. 
 
 Building on Kubernetes would mean that any CI/CD tool which has been Kubernetes-native built with building on Kubernetes as it’s main idea would thrive better than Jenkins itself. Leveraging this, we can use the capability of spinning pipelines on Tekton (Serverless CI/CD for Kubernetes) through Jenkins. Using the same vocabulary and ideas to execute Jobs from Jenkins and having the workloads of the Jobs run by Tekton in Kubernetes, this plugin aims to bridge the gap between Jenkins and Kubernetes.
 
-Jenkins currently does not support Tekton, making it hard for users to use it without prior knowledge of the CI/CD. This plugin would enable users to carry their previous Jenkins knowledge to translate it to Tekton.
+The Tekton Client Plugin allows users to create/delete resources via Build Steps but there is no way for users to employ their preexisting Jenkinsfile to extend their Jenkins pipeline to leverage Tekton for deploying on Kubernetes. 
 
-==== Project Details
-This project idea proposes to implement a Jenkins plugin which allows users to trigger pipelines in Tekton.
+With no DSL for Tekton on Jenkins there is no way for the users to write scripts which can automate their Tekton lifecycle through Jenkins.
 
+==== Project Details
+Leverage the Tekton command-line tool (tkn)[https://github.com/tektoncd/cli] to create a DSL which can be used in a groovy based Jenkinsfile to manage Tekton specific lifecycle.
```

Comment:
use an adoc link rather than a markdown link
```suggestion
Leverage the Tekton command-line tool link:https://github.com/tektoncd/cli[tkn] to create a DSL which can be used in a groovy based Jenkinsfile to manage Tekton specific lifecycle.
```